inject/inject/src/main/java/io/helidon/inject/InjectionServiceProviderException.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2022, 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.inject;
+
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * An exception relative to a {@link io.helidon.inject.ServiceProvider}.
+ */
+public class InjectionServiceProviderException extends InjectionException {
+
+    /**
+     * The service provider this exception pertains.
+     */
+    private final ServiceProvider<?> serviceProvider;
```

Comment:
```suggestion
    private final Optional<ServiceProvider<?>> serviceProvider;
```

Reply:
I do not agree with this one at all. We only need the provider instance for message, or when requested. 
The wrapping only introduces overhead here.

Reply:
np just a suggestion